packages/smart-wallet/test/test-marshal-contexts.js

```diff
@@ -75,16 +73,16 @@ test('makeImportContext preserves identity across AMM and wallet', t => {
   myWallet.suggestIssuer('ATOM', 'board011');
   const walletCapData = myWallet.publish();
   t.deepEqual(walletCapData, {
-    body: JSON.stringify([
+    body: `#${JSON.stringify([
```

Comment:
Obviously not to be addressed by this PR, but this nested JSON serialization remains as disturbing as ever. Do we have an issue for addressing it?

Are these nested serialization strings ever stored in durable storage? If so, will it become much harder to fix these later?

Reply:
Answering the above questions is not a precondition to proceeding on this PR.

Reply:
That case looks like the test is comparing the serialized forms, rather than parsing `walletCapData` and comparing the unserialized forms. As such, the `JSON.stringify` is being used as an ad-hoc `marshal.toCapData`, with manually-written `$0` Remotable references and a leading `#` prefix.

It seems that `myWallet.publish()` encapsulates a serializer, and this test is checking the results of calling that serializer (including the boardIDs that appear in `slots`). So it's kind of a toss-up as to whether it'd be better to stick with the current approach (which expresses the intent of the test much more directly), or to build a separate unserializer and do some form of `t.deepEqual(unserialize(walletCapData), { brand: ... }`. The problem with the second approach is that we really do care about boardIDs here, not just `iface`s, so we'd want the kind of unserializer that embeds the slot string into the `iface` to allow comparison.

While there are probably cases of nested serialization (i.e. `JSON.stringify(JSON.stringify(stuff))`), I don't think this is one of them.